### PR TITLE
[nvidia-bluefield] add sonic-byo python script

### PR DESCRIPTION
**Pull request by Yakiv-Huryk**
#### Why I did it
To add a possibility to disable SONiC containers and run a user-provided data-plane application.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Added Nvidia platform-specific script sonic-byo.py

#### How to verify it
Manual test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by oleksandrivantsiv — APPROVED** [review]

File: platform/nvidia-bluefield/byo/sonic-byo.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+
+def run_container(image):
+    print(f'# Running image {image}')
+    config = {
+        'image': image,
+        'name': CONTAINER_NAME,
+        'detach': True,
+        'tty': True,
+        'privileged': True,
```
Comment (maipbui) [inline platform/nvidia-bluefield/byo/sonic-byo.py:146]: >'privileged': True, [](http://example.com/codeflow?start=8&length=19)

running container with full privilege is dangerous, possible to limit the privileged scope?

File: platform/nvidia-bluefield/byo/sonic-byo.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    ]
+
+    systemctlops_start = ['unmask', 'enable', 'start']
+    systemctlops_stop = ['stop', 'disable', 'mask']
+
+    # Start featured last
+    if start:
+        services = services[1:] + services[:1]
+
+    print('# ' 'Starting' if start else 'Stopping', ', '.join(services))
```
Comment (maipbui) [inline platform/nvidia-bluefield/byo/sonic-byo.py:74]: >'# ' 'Starting' [](http://example.com/codeflow?start=10&length=15)

missing a comma? `'# ', 'Starting'`

Comment (Yakiv-Huryk) [reply platform/nvidia-bluefield/byo/sonic-byo.py]: thanks, fixed

File: platform/nvidia-bluefield/byo/sonic-byo.py (hunk 1 above)
Comment (Yakiv-Huryk) [reply platform/nvidia-bluefield/byo/sonic-byo.py:146]: The idea is that the container/app takes full control over the system, so full privilege is expected

**Comment by Yakiv-Huryk** [comment]
@maipbui can you please review?

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**